src/main/scala/org/broadinstitute/hail/driver/ExportVariantsCass.scala

```diff
@@ -271,7 +273,27 @@ object ExportVariantsCass extends Command {
                   .values(names, values))
               }
 
-            futures.foreach(_.getUninterruptibly())
+            futures.foreach( f => {
+              var retry = true
+              var retryInterval = 3 * 1000 // 3s
+
+              while (retry) {
+                try {
+                  f.getUninterruptibly()
```

Comment:
I don't think it is enough to get the Future again on retry, you need to re-execute the insert.  I'd do it like this:

Convert the block into a Seq of (names, values) pairs.  Note, there is a bug in Scala ArrayBuilder that in rare cases can cause it to reuse the underlying storage.  In this case, I'd use ArrayBuffers instead of ArrayBuilders and use `.toArray` to get the arrays out at the end.

The retry loop should run while the Seq of (names, values) pairs are non-empty.  In the retry loop, insert each remaining pair.  If a pair succeeds, filter it out.  If a pair fails, keep it for retry.

Make sense?